dapp/src/components/buySell/BalanceHeader.js

```diff
@@ -77,119 +73,31 @@ const BalanceHeader = () => {
-                  s.addOusdModalState = 'show'
-                })
-              }
-            },
-            // non even duration number so more of the decimals in ousdBalance animate
-            duration: 1985,
-            id: 'header-balance-ousd-animation',
-            stepTime: 30,
-          })
-        } else {
-          const storedAnimationData = localStorage.getItem('animatedBalance')
-          if (storedAnimationData) {
-            const animationData = JSON.parse(storedAnimationData)
-
-            /* OUSD balance has not changed, meaning no mint/redeem/transfer/rebase happened
-             * from the last time user opened the dapp
-             */
-            if (animationData.ousdBalance === ousdBalance) {
-              // time past since ousd animation last stored in seconds
-              const timePassed =
-                (Date.now() - Date.parse(animationData.time)) / 1000
-
-              /* increase the animated OUSD according to the apy and the
-               * time passed since last refresh.
-               *
-               * Important (!): this function does not account for APY changing and
-               * only takes the current APY into the account. Also if rebase happened
-               * 2 days ago, and user has not opened the app since, the current implementation
-               * is not able to simulate the balance increase since the rebase, and displays only the
-               * ousd balance in the amount immediately after rebase.
-               */
-              const simulatedOusdBalance =
-                parseFloat(animationData.animatedOusdBalance) +
-                ((parseFloat(animationData.animatedOusdBalance) * (apy || 0)) /
-                  // 31536000: amount of seconds in a year
-                  31536000) *
-                  timePassed
-              animateCancel = normalOusdAnimation(simulatedOusdBalance)
-            } else {
-              animateCancel = normalOusdAnimation(ousdBalance)
-            }
-          } else {
-            animateCancel = normalOusdAnimation(ousdBalance)
-          }
-        }
-      }, 10)
-
-      return () => {
-        clearTimeout(timeout)
-        if (animateCancel) animateCancel()
+      // user must have minted the OUSD
+      if (
+        typeof ousdBalanceNum === 'number' &&
+        typeof prevOusdBalanceNum === 'number' &&
+        Math.abs(ousdBalanceNum - prevOusdBalanceNum) > mintAnimationLimit
+      ) {
+        normalOusdAnimation(prevOusdBalance, ousdBalance)
+      } else if (
+        typeof ousdBalanceNum === 'number' &&
+        ousdBalanceNum > mintAnimationLimit
+      ) {
+        normalOusdAnimation(0, ousdBalance)
       }
     }
-  }, [ousdBalance, apy])
+  }, [ousdBalance])
 
-  const displayedBalance = formatCurrency(animatedOusdBalance || 0, 6)
-  const displayedBalanceNum = parseFloat(displayedBalance)
+  const displayedBalance = formatCurrency(animatedOusdBalance || 0, 2)
```

Comment:
Let's leave this at 6 decimals for now (with four of them grey).

Reply:
Done